### PR TITLE
Add option to account for chirality for structure searches

### PR DESCRIPTION
**Pull request by JJ-Pineda**
Resolves [#102](https://github.com/rdkit-rs/cheminee/issues/102) by adding the option to implement chiral substruct matches for the structure endpoints (i.e. identity, substructure, superstructure). The default is to ignore chirality.

### CODE REVIEW

File: src/command_line/search/cli_structure_search.rs [hunk 1]
```diff
@@ -12,11 +12,18 @@ pub fn cli_structure_search(method: &str, matches: &ArgMatches) -> eyre::Result<
...
         .get_one::<String>("smiles")
         .ok_or(eyre::eyre!("Failed to extract SMILES"))?;
+    let use_chirality = matches.get_one::<String>("use-chirality");
     let result_limit = matches.get_one::<String>("result-limit");
     let tautomer_limit = matches.get_one::<String>("tautomer-limit");
     let extra_query = matches.get_one::<String>("extra-query");
     let use_scaffolds = matches.get_one::<String>("use-scaffolds");
 
+    let use_chirality = if let Some(use_chirality) = use_chirality {
+        !matches!(use_chirality.as_str(), "false")
```
Comment (xrl) [inline src/command_line/search/cli_structure_search.rs:22]: Remind me, what does `matches!` do and where does it come from?

File: tests/structure_matching_tests.rs [hunk 2]
```diff
@@ -9,8 +9,8 @@ fn test_exact_match() {
     let romol1 = ROMol::from_smiles(smiles1).unwrap();
     let romol2 = ROMol::from_smiles(smiles2).unwrap();
 
-    assert!(exact_match(&romol1, &romol1.clone()));
-    assert!(!exact_match(&romol1, &romol2));
+    assert!(exact_match(&romol1, &romol1.clone(), true));
+    assert!(!exact_match(&romol1, &romol2, true));
```
Comment (xrl) [inline tests/structure_matching_tests.rs:13]: Can you make a test where you explore what happens in a scenario when use-chirality is true AND when it's false?

**Review by xrl — CHANGES_REQUESTED** [review]

File: src/command_line/search/cli_structure_search.rs (hunk 1 above)
Comment (JJ-Pineda) [reply src/command_line/search/cli_structure_search.rs:22]: `matches!(thing, "true")` is the equivalent of:

`match thing {`
`    "true" => true,`
`    _ => false,`
`}`

File: tests/structure_matching_tests.rs (hunk 2 above)
Comment (JJ-Pineda) [reply tests/structure_matching_tests.rs:14]: Good call! Went ahead and added a test to compare two molecules that match when use-chirality is set to false but fail to match when use-chirality is set to true.

**Review by xrl — APPROVED** [review]

**Pull request merged**